File: app/src/main/java/com/petnbu/petnbu/model/Paging.java
package com.petnbu.petnbu.model;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Ignore;
import android.arch.persistence.room.PrimaryKey;
import android.arch.persistence.room.TypeConverters;
import android.support.annotation.NonNull;

import com.petnbu.petnbu.db.PetTypeConverters;

import java.util.List;

@Entity(tableName = "paging")
@TypeConverters(PetTypeConverters.class)
public class Paging {

    public static final String GLOBAL_FEEDS_PAGING_ID = "global-feeds-paging-id-feeds";
    public static String feedCommentsPagingId(String feedId){
        return feedId + "-comments";
    }

    public static String subCommentsPagingId(String parentCmtId){
        return parentCmtId + "-subComments";
    }
    public static String userFeedsPagingId(String userId){
        return userId + "-feeds";
    }

    public static boolean isFeedPagingId(@NonNull String pagingId){
        return pagingId.endsWith("feeds");
    }

    public static boolean isCommentPagingId(@NonNull String pagingId){
        return pagingId.endsWith("comments") || pagingId.endsWith("subComments");
    }

    @PrimaryKey
    @NonNull
    private String pagingId;

    private List<String> ids;

    private boolean ended;

    private String oldestId;

    public Paging() {
    }

    @Ignore
    public Paging(@NonNull String pagingId, List<String> ids, boolean ended, String oldestId) {
        this.pagingId = pagingId;
        this.ids = ids;
        this.ended = ended;
        this.oldestId = oldestId;
    }

    public String getPagingId() {
        return pagingId;
    }

    public void setPagingId(String pagingId) {
        this.pagingId = pagingId;
    }

    public List<String> getIds() {
        return ids;
    }

    public void setIds(List<String> ids) {
        this.ids = ids;
    }

    public String getOldestId() {
        return oldestId;
    }

    public void setOldestId(String oldestId) {
        this.oldestId = oldestId;
    }

    public boolean isEnded() {
        return ended;
    }

    public void setEnded(boolean ended) {
        this.ended = ended;
    }

    public  void replaceId(String oldId, String newId){
        if(ids.size() == 1){
            ids.set(0, newId);
            this.oldestId = newId;
            return;
        }
        for (int i = 0; i < ids.size(); i++) {
            String id = ids.get(i);
            if (id.equals(oldId)) {
                ids.set(i, newId);
                return;
            }
        }
    }
}

File: app/src/main/java/com/petnbu/petnbu/model/Paging.kt
package com.petnbu.petnbu.model

import android.arch.persistence.room.Entity
import android.arch.persistence.room.Ignore
import android.arch.persistence.room.PrimaryKey
import android.arch.persistence.room.TypeConverters

import com.petnbu.petnbu.db.PetTypeConverters

@Entity(tableName = "paging")
@TypeConverters(PetTypeConverters::class)
class Paging {

    @PrimaryKey
    lateinit var pagingId: String

    private var ids: MutableList<String>? = null

    var isEnded: Boolean = false

    var oldestId: String? = null

    constructor()

    @Ignore
    constructor(pagingId: String, ids: MutableList<String>?, ended: Boolean, oldestId: String?) {
        this.pagingId = pagingId
        this.ids = ids
        this.isEnded = ended
        this.oldestId = oldestId
    }

    fun getIds() = ids

    fun setIds(ids: MutableList<String>) {
        this.ids = ids
    }

    companion object {

        val GLOBAL_FEEDS_PAGING_ID = "global-feeds-paging-id-feeds"

        val NOTIFICATION_PAGING_ID = "notifcation-paging-id"

        fun notificationsPagingId(): String {
            return NOTIFICATION_PAGING_ID
        }

        fun feedCommentsPagingId(feedId: String): String {
            return "$feedId-comments"
        }

        fun subCommentsPagingId(parentCmtId: String): String {
            return "$parentCmtId-subComments"
        }

        fun userFeedsPagingId(userId: String): String {
            return "$userId-feeds"
        }

        fun isFeedPagingId(pagingId: String): Boolean {
            return pagingId.endsWith("feeds")
        }

        fun isCommentPagingId(pagingId: String): Boolean {
            return pagingId.endsWith("comments") || pagingId.endsWith("subComments")
        }
    }
}
